test/cpp/util/cli_credentials.cc

```diff
@@ -22,27 +22,43 @@
 
 DEFINE_bool(enable_ssl, false, "Whether to use ssl/tls.");
 DEFINE_bool(use_auth, false, "Whether to create default google credentials.");
+DEFINE_string(access_token, "",
+              "The access token in metadata \"authorization\".");
 
 namespace grpc {
 namespace testing {
 
 std::shared_ptr<grpc::ChannelCredentials> CliCredentials::GetCredentials()
     const {
-  if (!FLAGS_enable_ssl) {
-    return grpc::InsecureChannelCredentials();
-  } else {
+  if (!FLAGS_access_token.empty()) {
     if (FLAGS_use_auth) {
-      return grpc::GoogleDefaultCredentials();
-    } else {
-      return grpc::SslCredentials(grpc::SslCredentialsOptions());
+      fprintf(
+          stderr,
+          "warning: use_auth is set to false when access_token is provided.");
     }
+
```

Comment:
You may also want to check `FLAGS_enable_ssl` is true here?

Reply:
`access_token` could only be used with the ssl credential here. The `enable_ssl` check may not be needed.

Reply:
What I was saying is that user can set `enable_ssl=false` with `access_token` and we still use ssl. This is not intuitive and we should log a warning or something.